### PR TITLE
fix code generation

### PR DESCRIPTION
**Pull request by flashdesignory**
if dataLayer is not passed in:

before:
Screenshot 2024-08-02 at 10 44 08 AM

after:
Screenshot 2024-08-02 at 10 44 33 AM

### CODE REVIEW

File: src/third-parties/google-analytics/data.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
       "key": "gtag"
     },
     {
-      "code": "window[{{l}}]=window[{{l}}]||[];window[{{l}}].push({'js':new Date()});window[{{l}}].push({'config':{{id}}})",
+      "code": "window[{{l}}]=window[{{l}}]||[];function gtag(){window[{{l}}].push(arguments);};gtag('js',new Date());gtag('config','{{id}}')",
```
Comment (flashdesignory) [inline src/third-parties/google-analytics/data.json:18]: just pushing init code in the dataLayer doesn't work. similar to the original implementation we are using a gtag function. We are just not making it global by NOT assigning it to the windows object.

File: src/utils/index.ts [hunk 2]
```diff
@@ -57,7 +57,7 @@ export function formatCode(
   return code.replace(/{{(.*?)}}/g, (match) => {
     const name = match.split(/{{|}}/).filter(Boolean)[0];
     return JSON.stringify(
-      args && name in args ? args?.[name] : optionalParams?.[name],
+      args?.[name] !== undefined ? args?.[name] : optionalParams?.[name],
```
Comment (flashdesignory) [inline src/utils/index.ts:60]: if `l` wasn't assigned, we returned `undefined`

File: src/third-parties/google-analytics/data.json (hunk 1 above)
Comment (huang-julien) [reply src/third-parties/google-analytics/data.json:18]: `gtag` was removed because having multiple gtm/ga would override the last `gtag`

Comment (flashdesignory) [reply src/third-parties/google-analytics/data.json:18]: it's not on windows. we can also use any function name we like, ie:

window['gtag-{{l}}] = function(){}

Comment (flashdesignory) [reply src/third-parties/google-analytics/data.json:18]: well... let me use the dynamic approach...uno momento

File: src/utils/index.ts [hunk 3]
```diff
@@ -56,9 +56,7 @@ export function formatCode(
 ) {
   return code.replace(/{{(.*?)}}/g, (match) => {
     const name = match.split(/{{|}}/).filter(Boolean)[0];
-    return JSON.stringify(
-      args && name in args ? args?.[name] : optionalParams?.[name],
-    );
+    return args?.[name] !== undefined ? args?.[name] : optionalParams?.[name];
```
Comment (huang-julien) [inline src/utils/index.ts:59]: so we d'ont stringify it anymore ? would this be a breaking change ?

Comment (huang-julien) [reply src/utils/index.ts:59]: maybe not, since all data json are modified. But how should null be handled ?

Comment (flashdesignory) [reply src/utils/index.ts:59]: it's in the code block now, cause this was returning window['undefined'] otherwise if 'l' wasn't set

**Review by housseindjirdeh — APPROVED** [review]

File: src/utils/index.ts (hunk 3 above)
Comment (housseindjirdeh) [reply src/utils/index.ts:59]: Nit, but wouldn't both a null + undefined check be better here?

```suggestion
    return args[name] ??  optionalParams?.[name];
```

Comment (flashdesignory) [reply src/utils/index.ts:59]: i'm not sure what's better... in theory an arg could have a key with a null value on purpose?

Comment (huang-julien) [reply src/utils/index.ts:59]: I guess that a null value is an intentional value set from users 🤔

**Review by huang-julien — APPROVED** [review]

**Pull request merged**